### ISSUE TITLE
linter: Invalid fix for eslint/prefer-const rule

### ISSUE THREAD

**Issue by connorshea** [oxc-project/oxc#18810]
### What version of Oxlint are you using?

main

### What does your `.oxlintrc.json` config file look like?

N/A

### What happened?

Noticed this failure in the ecosystem-ci: https://github.com/oxc-project/oxc-ecosystem-ci/actions/runs/21565097141/job/62135469720

Basically, this code is transformed into invalid syntax:

Input:
```js
export const getStaticProps = ({ params }) => {
  if (params.post.startsWith('redir')) {
    let destination = '/404'

    if (params.post.includes('dest-external')) {
      destination = 'https://example.vercel.sh'
    } else if (params.post.includes('dest-')) {
      destination = params.post.split('dest-').pop().replace(/_/g, '/')
    }

    let permanent = undefined
    let statusCode = undefined

    if (params.post.includes('statusCode-')) {
      permanent = parseInt(
        params.post.split('statusCode-').pop().split('-').shift(),
        10
      )
    }

    if (params.post.includes('permanent')) {
      permanent = true
    } else if (!statusCode) {
      permanent = false
    }

    const redirect = {
      destination,
      permanent,
      statusCode,
    }

    if (params.post.includes('no-basepath-')) {
      redirect.basePath = false
    }

    return { redirect }
  }

  return {
    props: {
      params,
    },
  }
}
```

Output:
```js
export const getStaticProps = ({ params }) => {
  if (params.post.startsWith('redir')) {
    let destination = '/404'

    if (params.post.includes('dest-external')) {
      destination = 'https://example.vercel.sh'
    } else if (params.post.includes('dest-')) {
      destination = params.post.split('dest-').pop().replaceAll(/_/g, '/')
    }

    let permanent
    const statusCode

    if (params.post.includes('statusCode-')) {
      permanent = Number.parseInt(
        params.post.split('statusCode-').pop().split('-').shift(),
        10
      )
    }

    if (params.post.includes('permanent')) {
      permanent = true
    } else if (!statusCode) {
      permanent = false
    }

    const redirect = {
      destination,
      permanent,
      statusCode,
    }

    if (params.post.includes('no-basepath-')) {
      redirect.basePath = false
    }

    return { redirect }
  }

  return {
    props: {
      params,
    },
  }
}
```

This is seemingly a mistake caused by the fixer for prefer-const and its interaction with `no-useless-undefined`.

<details>
<summary>Full log from ecosystem-ci</summary>

````md
thread '<unnamed>' (2697) panicked at crates/oxc_linter/src/fixer/mod.rs:409:13:
Linter fixer produced invalid syntax.

Input code: 
```
import { useRouter } from 'next/router'

export default function Post(props) {
  const router = useRouter()

  if (typeof window === 'undefined') {
    if (router.query.post?.startsWith('redir')) {
      console.log(router)
      throw new Error('render should not occur for redirect')
    }
  }

  if (typeof window !== 'undefined' && !window.initialHref) {
    window.initialHref = window.location.href
  }

  if (router.isFallback) return <p>Loading...</p>

  return (
    <>
      <p id="gsp">getStaticProps</p>
      <p id="props">{JSON.stringify(props)}</p>
    </>
  )
}

export const getStaticProps = ({ params }) => {
  if (params.post.startsWith('redir')) {
    let destination = '/404'

    if (params.post.includes('dest-external')) {
      destination = 'https://example.vercel.sh'
    } else if (params.post.includes('dest-')) {
      destination = params.post.split('dest-').pop().replace(/_/g, '/')
    }

    let permanent = undefined
    let statusCode = undefined

    if (params.post.includes('statusCode-')) {
      permanent = parseInt(
        params.post.split('statusCode-').pop().split('-').shift(),
        10
      )
    }

    if (params.post.includes('permanent')) {
      permanent = true
    } else if (!statusCode) {
      permanent = false
    }

    const redirect = {
      destination,
      permanent,
      statusCode,
    }

    if (params.post.includes('no-basepath-')) {
      redirect.basePath = false
    }

    return { redirect }
  }

  return {
    props: {
      params,
    },
  }
}

export const getStaticPaths = () => {
  return {
    paths: ['first', 'second'].map((post) => ({ params: { post } })),
    fallback: true,
  }
}

```

Fixed code: 
```
import { useRouter } from 'next/router'

export default function Post(props) {
  const router = useRouter()

  if (typeof window === 'undefined') {
    if (router.query.post?.startsWith('redir')) {
      console.log(router)
      throw new Error('render should not occur for redirect')
    }
  }

  if (typeof window !== 'undefined' && !window.initialHref) {
    window.initialHref = window.location.href
  }

  if (router.isFallback) {return <p>Loading...</p>}

  return (
    <>
      <p id="gsp">getStaticProps</p>
      <p id="props">{JSON.stringify(props)}</p>
    </>
  )
}

export const getStaticProps = ({ params }) => {
  if (params.post.startsWith('redir')) {
    let destination = '/404'

    if (params.post.includes('dest-external')) {
      destination = 'https://example.vercel.sh'
    } else if (params.post.includes('dest-')) {
      destination = params.post.split('dest-').pop().replaceAll(/_/g, '/')
    }

    let permanent
    const statusCode

    if (params.post.includes('statusCode-')) {
      permanent = Number.parseInt(
        params.post.split('statusCode-').pop().split('-').shift(),
        10
      )
    }

    if (params.post.includes('permanent')) {
      permanent = true
    } else if (!statusCode) {
      permanent = false
    }

    const redirect = {
      destination,
      permanent,
      statusCode,
    }

    if (params.post.includes('no-basepath-')) {
      redirect.basePath = false
    }

    return { redirect }
  }

  return {
    props: {
      params,
    },
  }
}

export const getStaticPaths = () => ({
    paths: ['first', 'second'].map((post) => ({ params: { post } })),
    fallback: true,
  })

```

Parse errors: [OxcDiagnostic { inner: OxcDiagnosticInner { message: "Missing initializer in const declaration", labels: Some([LabeledSpan { label: None, span: SourceSpan { offset: SourceOffset(955), length: 10 }, primary: false }]), help: Some("Add an initializer (e.g. ` = undefined`) here"), note: None, severity: Error, code: OxcCode { scope: None, number: None }, url: None } }]
````
</details>

**Comment by connorshea** [comment]
This might be a bad interaction between two fixers? Not sure.

I tried adding a fix test, but it passes fine:

```rs
(
    "const f = () => {
        let a = undefined
        let b = undefined

        if (foo === 'bar') {
          a = parseInt(baz, 10)
        }

        if (foo > bar) { a = true }

        const c = { a, b }

        return { c }
      }",
    "const f = () => {
        let a = undefined
        const b = undefined

        if (foo === 'bar') {
          a = parseInt(baz, 10)
        }

        if (foo > bar) { a = true }

        const c = { a, b }

        return { c }
      }",
    None,
),
```

**Comment by connorshea** [comment]
Yeah it's probably a conflict with also running `unicorn/no_useless_undefined`, not sure how best to fix that...

**Comment by connorshea** [comment]
cc @camc314 if you have any ideas on how we've fixed this kind of problem in the past.

**Comment by camc314** [comment]
We can change the span of the fixer for prefer_const to replace the whole stmt rather than just the keyword

**Comment by connorshea** [comment]
Confirmed that the panic is fixed 👍 https://github.com/oxc-project/oxc-ecosystem-ci/actions/runs/21571457697/job/62151429886
